src/sentry/tasks/store.py

```diff
@@ -505,8 +507,13 @@ def _do_save_event(
 
     event_type = "none"
 
+    if consumer_type and consumer_type == ConsumerType.Transactions:
```

Comment:
how come it's optional? imo we should always pass it in, it's less risky to be stricter about types

```suggestion
    if consumer_type == ConsumerType.Transactions:
```